packages/backend-core/src/auth/auth.ts

```diff
@@ -188,3 +192,32 @@ export async function updateUserOAuth(userId: string, oAuthConfig: any) {
     console.error("Could not update OAuth details for current user", e)
   }
 }
+
+/**
+ * Logs a user out from budibase. Re-used across account portal and builder.
+ */
+export async function platformLogout(opts: PlatformLogoutOpts) {
+  const ctx = opts.ctx
+  const userId = opts.userId
+  const keepActiveSession = opts.keepActiveSession
+
+  if (!ctx) throw new Error("Koa context must be supplied to logout.")
+
+  const currentSession = getCookie(ctx, Cookie.Auth)
+  let sessions = await getSessionsForUser(userId)
+
+  if (keepActiveSession) {
```

Comment:
What is the use case of keeping the session active on logout?

Reply:
The only current use case is updating your own password from inside the builder

https://github.com/Budibase/budibase/blob/c4ef5b30bf56e7cca987116def5aa24afeb31ba7/packages/worker/src/api/controllers/global/self.ts#L137-L142